tests/evm/test_create.py

```diff
@@ -209,20 +226,33 @@ def test_create_create2(
         contract_addr = keccak256(rlp.encode([caller.address.to_bytes(20, "big"), caller.nonce]))
     contract_address = int.from_bytes(contract_addr[-20:], "big")
 
-    # can't be a static all
-    is_static = 0
-
-    next_call_id = 65
+    next_call_id = 66
     rw_counter = next_call_id
+
     # CREATE: 33 * 3(push) + 1(CREATE) + 1(mstore) + 33(PUSH32) + 2(PUSH) + 1(RETURN)
     # CREATE2: 33 * 4(push) + 1(CREATE2) + 1(mstore) + 33(PUSH32) + 2(PUSH) + 1(RETURN)
-    next_program_counter = 33 * 4 + 1 + 35 + 1 if is_create2 else 33 * 3 + 1 + 35 + 1
-    assert caller_bytecode.code[next_program_counter] == opcode
+    pc = 33 * 4 + 1 if is_create2 else 33 * 3 + 1
+    if has_init_code:
+        next_program_counter = pc + 1 + 35 + 1 if is_create2 else pc + 1 + 35 + 1
+    else:
+        next_program_counter = pc
```

Comment:
I think we can simplify this
```suggestion
    next_program_counter = 33 * 4 + 1 if is_create2 else 33 * 3 + 1
    if has_init_code:
        next_program_counter += 1 + 35 + 1
```